sanityio-case-study-bettytro.md (#225)
NoroffFEU/development-platforms: src/content/docs/baas/sanity/sanityio-case-study-bettytro.md

---
title: Sanity.io - Headless CMS
keywords: development platforms, sanity, headlesscms
tags: Development Platforms, Sanity, Headless CMS
---

## Introduction

> This assessment will cover my experiences with sanity.io, I will also go in to headless CMS a bit. There is no understanding Sanity without understanding the basics of headless CMS. I will also reflect on why I think you should pick a headless CMS, and why it probably should be Sanity.

Reviewer: Technical writing does not use I/me as the idea is not to give your opinion but factual details on the subject.

```suggestion
In this case study we will assess sanity.io, as well as discussing the concept of headless CMS...
```

Then it hit me… I am currently just building the back-end, and as a frontend developer I prefer to see what I build with graphics, not just queries with GROQ in the «Vision Tool» in the Studio. Don’t mistake my love for graphics as a hate for the Vision tool, the tool is brilliant, and better than most query tools I have ever tested. This is also because GROQ is brilliant, and really easy to learn. 

But here I was, deciding to build a front for this project that I still didn’t know anything about. I had no plan for the project, I just wanted to learn, so how could I decide on a front for it? I thought about just doing what I know already, maybe just using HTML, CSS and Portable text from Sanity. I thought about React, as the Studio is built in it, and because I know some React already. I did some research and it seemed like most people prefer to use Next.js with Sanity, so that is what I decided with. The server-side rendering in Next is perfect for the queries, and building components in Next is basically React. 

> For me, being able to see the effects of the schemas I write is key. I can change them in a few seconds if I come up with some other field I need in that «document». Building a schema for a blog and I realize I need a field for a slug so I can create nice looking URLs? Easy. I just opened up my schema and added a field. Realizing I can’t have a blog post without SEO fields? Create a new field set, add a field for meta description, add an array of keywords. Oh, forgot about OpenGraph for a second, just a few more lines of code and everything is in place. That is what impressed me the most about Sanity. The highly customizable Studio that easily can be created and changed like you want it. 

Reviewer: ```suggestion
For me, being able to see the effects of the schemas I write is key. I can change them in a few seconds if I come up with some other field I need in that "document". Building a schema for a blog and I realize I need a field for a slug so I can create nice looking URLs? Easy. I just opened up my schema and added a field. Realizing I can’t have a blog post without SEO fields? Create a new field set, add a field for meta description, add an array of keywords. Oh, forgot about OpenGraph for a second, just a few more lines of code and everything is in place. That is what impressed me the most about Sanity. The highly customizable Studio that easily can be created and changed like you want it. 
```

Please use international quotes.

3. **Powerful querying language:** Sanity uses a unique querying language called GROQ that allows developers to retrieve their data in the exact structure they need it. This can make it easier to work in complex data structures and can lead to more efficient data retrieval.

4. **Developer-friendly:** Sanity is built with a development-centric approach.

In conclusion, Sanity offers a range of features that set it apart from other headless CMS platforms. Their hosting is also cheap, and their CDNs are fast and reliable. There are plenty of plugins and help tools available as well, and there is plenty of documentation. 

This combined with the customizable nature of the open source studio, and the easy schema structure are some of the reasons that Sanity is preferred and used by a lot of big customers and developer teams. With customers like Nike, Puma and Shopify, and a whopping 20 billion API / CDN requests every month, we can easily say that Sanity is popular, and that is for a reason.

### Credits

> - Elisabeth Trondsen (github.com/bettytro)

Reviewer: You should turn this into a proper link
```suggestion
- [Elisabeth Trondsen](github.com/bettytro)
```

### References

> sanity.io/docs

Reviewer: Same here, should be a link
